SETI/rms-oops
# Move solar and polymath to separate repos. Move hosts under oops.

**Major restructuring:**

- Move `hosts/solar` to new pip-installable `rms-solar` repo.
- Move `polymath` to new pip-installable `rms-polymath` repo. Fixes #100.
- Move `hosts` directory under `oops` and change all references as appropriate. Fixes #78.

**Changes to testing:**

- Fixes #64  (changed test directory warning to include resolved paths; other complaints had already been fixed)
- Remove existing `backplane/unittester.py` since it was really just a Cassini ISS hosts test, and put it under `hosts/cassini` for now. At some point we will want to restructure the Cassini hosts module to follow the same format as Galileo SSI.
- Remove contents of `backplane/__init__.py` for the same reason.
- Because the module names now start with `oops.hosts` instead of just `hosts` I had to create new gold master directories in Dropbox (just copies of the existing ones). Once this pull is merged, we can delete the old directories.
- I also made some changes to the gold master system:
  - Output directories are only created when needed. If there are no files being written, no directories are created.
  - There are new options `--arrays` and `--no-arrays` that control writing of the backplane arrays.
  - There are new options `--summary` and `--no-summary` that control writing of the `summary.py` file.
  - Fixed various bugs in argument definition.
  - By default, nothing is written (no arrays, browse, log, or summary). There is a new option `--debug` which turns on `--arrays --browse --log --summary`. You may hate this, in which case we can revert to the original behavior, but I thought that by default running a test suite should not create a bunch of files.

**Minor changes:**

- Update `requirements.txt` to use new versions of modules and new modules above. Tested with Python 3.8 and 3.12.
- Update `.gitignore` to be the GitHub-standard template with our local additions.
- Move `.coveragerc` to the top-level directory and clean up as needed.
- Remove `oops/.vscode` which shouldn't be under source code control in the first place.

**Known problems:**

- Because the majority of `oops` unit tests include some combination of:

```
        Path.reset_registry()
        Frame.reset_registry()
        Body.reset_registry()
```

you can't currently mix hosts and non-hosts tests in the same unittest run. Since the order of tests can't be specified, the registries are emptied and then a host needs those entries that were previously loaded. So for now you have to run, separately:

```
python oops/unittester.py
python oops/hosts/unittester.py
```

Doing this, all tests pass.
